lusidtools/cocoon/instruments.py

```diff
@@ -292,17 +292,28 @@ def get_unique_identifiers(api_factory: lusid.utilities.ApiClientFactory):
         if identifier.is_unique_identifier_type
     ]
 
+def get_leaf(d, IDs):
+    for lusidID, dataID in d.items():
```

Comment:
Can you please add type annotations and a docstring to this function?

Reply:
You may find utilities.handle_nested_default_and_column_mapping interesting to look at (and potentially re-use) here.